browser/psst/psst_tab_web_contents_observer.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/psst/psst_tab_web_contents_observer.h"
+
+#include <memory>
+#include <utility>
+
+#include "brave/components/psst/browser/content/psst_scripts_result_handler.h"
+#include "brave/components/psst/common/features.h"
+#include "brave/components/psst/common/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/navigation_entry.h"
+#include "content/public/browser/navigation_handle.h"
+
+namespace psst {
+
+// static
+std::unique_ptr<PsstTabWebContentsObserver>
+PsstTabWebContentsObserver::MaybeCreateForWebContents(
+    content::WebContents* contents,
+    Profile* profile,
+    std::unique_ptr<PsstDialogDelegate> delegate,
+    const int32_t world_id) {
+  if (!profile || contents->GetBrowserContext()->IsOffTheRecord() ||
```

Comment:
profile cannot be null here so either `CHECK` or just skip it. Also there's no reason to call `contents->GetBrowserContext()` because you already have `profile`

Reply:
actually the problem here is passing in Profile at all, you should pass in `Prefs` instead just like `PsstTabWebContentsObserver` and then this can also be moved into the component